internal/conductorr/integration/util.go

```diff
@@ -0,0 +1,65 @@
+package integration
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+/*
+ * Follow a download URL, stopping when we reach a magnet URL or when we
+ * find a torrent file.
+ *
+ * Either:
+ * - We find a magnet URL: nextURL.Scheme == "magnet", and *respBuf should
+ *   be discarded, or,
+ * - We find a torrent file: nextURL.Scheme != "magnet", and *respBuf
+ *   contains the torrent file.
+ */
+func FollowDownloadURL(nextURL *url.URL, respBuf *[]byte) error {
```

Comment:
refactored. the function is a bit awkward. in a language with tagged unions, i'd return that but i don't believe go has that.

i also don't know how memory management works in go, so I don't know how cursed this function signature is.

in C, i'd do something like `char* follow_download_url(*URL next_url)` and have caller free the response buffer allocated in this function (since it seems like  `io.ReadAll` heap allocates) .